snowpack/src/sources/local.ts

```diff
@@ -202,125 +345,111 @@ export default {
+      };
+      await writeLockfile(path.join(config.root, LOCKFILE_NAME), newLockfile);
+      // Retry.
+      return this.installPackage(packageName, source);
     }
+    if (existsSync(arbNode.path)) {
+      return false;
+    }
+    await pacote.extract(arbNode.resolved, arbNode.path);
+    return true;
+  }
 
+  private async buildPackageImport(spec: string, _source?: string, logLine = false, depth = 0) {
+    const {config, memoizedResolve, allKnownSpecs, allPackageImports} = this;
+    const source = _source || this.packageRootDirectory;
     const aliasEntry = findMatchingAliasEntry(config, spec);
     if (aliasEntry && aliasEntry.type === 'package') {
       const {from, to} = aliasEntry;
       spec = spec.replace(from, to);
     }
 
-    const entrypoint = resolveEntrypoint(spec, {
-      cwd: path.dirname(source),
-      packageLookupFields: [
-        'snowpack:source',
-        ...((_config.packageOptions as PackageSourceLocal).packageLookupFields || []),
-      ],
-    });
     const specParts = spec.split('/');
     let _packageName: string = specParts.shift()!;
     if (_packageName?.startsWith('@')) {
       _packageName += '/' + specParts.shift();
     }
-    const isSymlink = !entrypoint.includes(path.join('node_modules', _packageName));
-    const isWithinRoot = config.workspaceRoot && entrypoint.startsWith(config.workspaceRoot);
-    if (isSymlink && config.workspaceRoot && isWithinRoot) {
-      const builtEntrypointUrls = getBuiltFileUrls(entrypoint, config);
-      const builtEntrypointUrl = slash(
-        path.relative(config.workspaceRoot, builtEntrypointUrls[0]!),
-      );
-      allSymlinkImports[builtEntrypointUrl] = entrypoint;
-      return path.posix.join(config.buildOptions.metaUrlPath, 'link', builtEntrypointUrl);
-    } else if (isSymlink && config.workspaceRoot !== false && !hasWorkspaceWarningFired) {
-      hasWorkspaceWarningFired = true;
-      logger.warn(
-        colors.bold(`${spec}: Locally linked package detected outside of project root.\n`) +
-          `If you are working in a workspace/monorepo, set your snowpack.config.js "workspaceRoot" to your workspace\n` +
-          `directory to take advantage of fast HMR updates for linked packages. Otherwise, this package will be\n` +
-          `cached until its package.json "version" changes. To silence this warning, set "workspaceRoot: false".`,
-      );
-    }
 
-    if (importMap) {
-      if (importMap.imports[spec]) {
-        return path.posix.join(config.buildOptions.metaUrlPath, 'pkg', importMap.imports[spec]);
+    // Before doing anything, check for symlinks because symlinks shouldn't be built.
+    try {
+      const entrypoint = resolveEntrypoint(spec, {
+        cwd: source,
+        packageLookupFields: [
+          'snowpack:source',
+          ...((config.packageOptions as PackageOptionsLocal).packageLookupFields || []),
+        ],
+      });
+      const isSymlink = !entrypoint.includes(path.join('node_modules', _packageName));
```

Comment:
Random question: why are some variables prefixed with `_`?

Reply:
yea It's a bit overloaded at the moment. Two reasons:

1. When you do this, TS will stop complaining about an "unused arg"
2. I'll also do this to represent an arg that needs to be converted in some way. ex: 

```
function foo(_packageName: string) {
  const packageName = _packageName ? _packageName.toLowerCase() : 'default';
}
```